File: 747. Largest Number At Least Twice of Others/747.go
package leetcode

func dominantIndex(nums []int) int {
    if len(nums)<2{
		return 0
	}

	var(
		b int
		a int
		index int
	)
	if nums[0]<nums[1]{
		a = nums[0]
		b = nums[1]
		index = 1
	}else{
		a = nums[1]
		b = nums[0]
		index = 0
	}

	for i:=2; i<len(nums); i++{
		if nums[i]>b{
			a, b = b, nums[i]
			index = i
		}else if nums[i]>a{
			a = nums[i]
		}
	}

	if b >= 2*a{
		return index
	}
	return -1
}

File: 747. Largest Number At Least Twice of Others/747.c
#include<stdio.h>

int dominantIndex(int* nums, int numsSize){
    int a,b,index;

    if (numsSize<2){
        return 0;
    }

    if (*nums<*(nums+1)){
        a = *nums;
        b = *(nums+1);
        index = 1;
    }else{
        b = *nums;
        a = *(nums+1);
        index = 0;
    }

    for (int i=2; i<numsSize; i++){
        if (*(nums+i)>b){
            int tmp = b;
            b = nums[i];
            a = tmp;
            index = i;
        }else if (*(nums+i)>a){
            a = nums[i];
        }
    }

    if (b >= 2*a){
        return index;
    }
    return -1;
}

int main(){
    int nums[4] = {3, 6, 1, 0};
    printf("%d", dominantIndex(nums, 4));
    return 0;
}